github/pull_request.rb

```diff
@@ -145,6 +154,16 @@ def add_status(state, message, options = {})
     @client.create_status(repo.full_name, options[:sha], state, context: 'prprocessor', description: message, target_url: options[:url])
   end
 
+  def set_directory_labels(mapping)
+    files = client.pull_files(repo.full_name, number)
+    desired_labels = files.collect { |f| mapping[f.filename.split("/").first] }.reject { |l| l.nil? }.unique
```

Comment:
I find this a bit complex with the multiple calls, and the use of single variable names.

Reply:
In the past year I've learned a bit more ruby. I think the ```.reject``` can be replaced with ```.compact```. Would that simplify it enough?

Reply:
Aye

Reply:
Updated